File: app/src/main/java/team/triplog/presentation/adapter/TripLogInfoViewPagerAdapter.java
package team.triplog.presentation.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;

import team.triplog.R;
import team.triplog.domain.model.TripLogInfo;

public class TripLogInfoViewPagerAdapter extends RecyclerView.Adapter<TripLogInfoViewPagerAdapter.MyViewHolder> {
    private Context context;
    private ArrayList<TripLogInfo> tripLogInfos;

    public TripLogInfoViewPagerAdapter(Context context, ArrayList<TripLogInfo> tripLogInfos) {
        this.context = context;
        this.tripLogInfos = tripLogInfos;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.item_trip_log_info_viewpager, parent, false);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        TripLogInfo tripLog = tripLogInfos.get(position);

    }

    @Override
    public int getItemCount() {
        return tripLogInfos.size();
    }

    public class MyViewHolder extends RecyclerView.ViewHolder {

        View view;

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
            view = itemView.findViewById(R.id.view);
        }
    }

}

File: app/src/main/java/team/triplog/presentation/adapter/TripLogInfoViewPagerAdapter.kt
package team.triplog.presentation.adapter

import android.content.Context
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import team.triplog.domain.model.TripLogInfo
import team.triplog.presentation.viewholder.TripLogInfoViewPagerViewHolder
import java.util.*

class TripLogInfoViewPagerAdapter(
    private val context: Context,
    private val items: ArrayList<TripLogInfo>
) : RecyclerView.Adapter<TripLogInfoViewPagerViewHolder>() {
    override fun getItemCount(): Int = items.size

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TripLogInfoViewPagerViewHolder {
        return TripLogInfoViewPagerViewHolder.getViewHolder(parent)
    }

    override fun onBindViewHolder(holder: TripLogInfoViewPagerViewHolder, position: Int) {
        val item = items[position]
        holder.bind(item)
    }
}